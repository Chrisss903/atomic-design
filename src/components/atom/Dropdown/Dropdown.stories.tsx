import Dropdown from "./Dropdown";

export default {
  title: "DropDown",
  component: Dropdown,
};

export const Primary_Dropdown = () => {
  const countries: string[] = [
    "India",
    "United States",
    "Germany",
    "Canada",
    "Australia",
    "Japan",
    "Brazil",
    "United Kingdom",
    "France",
    "South Africa",
  ];

  return <Dropdown children={"select the Country"} CountryList={countries} />;
};
