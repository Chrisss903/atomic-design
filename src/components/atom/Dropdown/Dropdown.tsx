import {
  Box,
  InputLabel,
  MenuItem,
  Select,
  SelectChangeEvent,
} from "@mui/material";

type DropdownProps = {
  children?: string;
  label?: string;
  CountryList?: string[];
  name?: string;
  value?: string; // Updated to include selected value
  onChange?: (event: SelectChangeEvent<string>) => void;
};

const Dropdown = ({
  children,
  CountryList,
  name,
  onChange,
  value,
}: DropdownProps) => {
  return (
    <Box
      display="flex"
      flexDirection={"column"}
      justifyContent={"center"}
      alignItems={"flex-start"}
      gap={2}
    >
      <InputLabel data-testid="Label" id="formCountry">
        {children}
      </InputLabel>
      <Select
        data-testid="Selectbar"
        onChange={onChange}
        value={value} // Make sure the value is set properly
        name={name}
        labelId="formCountry"
        sx={{ width: "14.3rem" }}
      >
        {CountryList &&
          CountryList.map((country, index) => (
            <MenuItem key={index} value={country} data-testid="Select-Option">
              {country}
            </MenuItem>
          ))}
      </Select>
    </Box>
  );
};

export default Dropdown;
