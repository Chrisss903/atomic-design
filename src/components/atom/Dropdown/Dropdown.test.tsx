import Dropdown from "./Dropdown";
import { fireEvent, render, screen } from "@testing-library/react";

const countries = [
  "India",
  "United States",
  "Germany",
  "Canada",
  "Australia",
  "Japan",
  "Brazil",
  "United Kingdom",
  "France",
  "South Africa",
];

test("Dropdown label is rendering", () => {
  render(<Dropdown CountryList={countries} children="Select Country" />);
  expect(screen.getByTestId("Label")).toBeInTheDocument();
});

test("Dropdown select bar is rendering", () => {
  render(<Dropdown CountryList={countries} />);
  expect(screen.getByTestId("Selectbar")).toBeInTheDocument();
});

test("Selecting an option updates the dropdown value-India", () => {
  render(<Dropdown CountryList={countries} />);

  const dropdownSelect = screen.getByRole("combobox");

  fireEvent.mouseDown(dropdownSelect);

  const option = screen?.getAllByTestId("Select-Option");
  fireEvent.click(option[0]);

  const Inputlabel1 = screen.getByDisplayValue("India");

  expect(Inputlabel1)?.toHaveValue("India");
});

test("Selecting an option updates the dropdown value-India", () => {
  render(<Dropdown CountryList={countries} />);

  const dropdownSelect = screen.getByRole("combobox");

  fireEvent.mouseDown(dropdownSelect);

  const option = screen?.getAllByTestId("Select-Option");
  fireEvent.click(option[1]);

  const Inputlabel1 = screen.getByDisplayValue("United States");

  expect(Inputlabel1)?.toHaveValue("United States");
});
