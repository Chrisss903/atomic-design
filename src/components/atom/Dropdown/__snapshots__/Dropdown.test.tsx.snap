// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown is Rendering 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1embff6"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1d2t4i-MuiFormLabel-root-MuiInputLabel-root"
      data-testid="Label"
      id="formCountry"
    />
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-3ivbi-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="Selectbar"
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="formCountry"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-15k6ek6-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1u6jos5-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-gpjuor"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown is Rendering 2`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1embff6"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1d2t4i-MuiFormLabel-root-MuiInputLabel-root"
      data-testid="Label"
      id="formCountry"
    />
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-3ivbi-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="Selectbar"
    >
      <div
        aria-controls=":r1:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="formCountry"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-15k6ek6-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1u6jos5-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-gpjuor"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown label is rendering 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-15vax0v"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1d2t4i-MuiFormLabel-root-MuiInputLabel-root"
      data-testid="Label"
      id="formCountry"
    />
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-1a03az7-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="Selectbar"
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="formCountry"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-15k6ek6-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1u6jos5-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-gpjuor"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown select bar is rendering 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-15vax0v"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1d2t4i-MuiFormLabel-root-MuiInputLabel-root"
      data-testid="Label"
      id="formCountry"
    />
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-1a03az7-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="Selectbar"
    >
      <div
        aria-controls=":r1:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="formCountry"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-15k6ek6-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1u6jos5-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-gpjuor"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
